Sources/main.cpp

```diff
@@ -19,6 +22,33 @@
 #include "core/FastIsType.h"
 #include "AutogenMetadata.h"
 
+QFile logs("error.log");
+
+void myMessageOutput(QtMsgType type, const QMessageLogContext &context, const QString &msg)
+{
+    QTextStream logstream(&logs);
+    QString datetime = QDateTime::currentDateTime().toString();
+    switch (type) {
+    case QtDebugMsg:
+        logstream << "Debug, " << datetime << ": " << msg << " " << context.file << " " << context.line << " " << context.function << endl;
+        break;
+    case QtWarningMsg:
+        logstream << "Warning, " << datetime << ": " << msg << " " << context.file << " " << context.line << " " << context.function << endl;
+        break;
+    case QtCriticalMsg:
+        logstream << "Critical, " << datetime << ": " << msg << " " << context.file << " " << context.line << " " << context.function << endl;
+        break;
+    case QtFatalMsg:
+        logstream << "Fatal, " << datetime << ": " << msg << " " << context.file << " " << context.line << " " << context.function << endl;
+        abort();
+#if QT_VERSION >= QT_VERSION_CHECK(5, 5, 0)
+    case QtInfoMsg:
```

Comment:
`QtInfoMsg` is out of order. http://doc.qt.io/qt-5/qtglobal.html#QtMsgType-enum - here you can see the canonical order in the enum.